commit 82f17fbe68d3204932e72296fe07fd416aa8f9fc
Author: Roman Bogorodskiy <bogorodskiy@gmail.com>
Date:   Thu Feb 4 03:13:10 2016 +0300

    configure: zfs: enable on Linux
    
    ZFS-on-Linux implementation of ZFS starting with version 0.6.4
    contains all the features we use. Additionally, as we support
    'volmode' option handling that's not available on ZoL but is
    available on FreeBSD, there is no need to block ZFS storage driver
    on Linux anymore.
    
    So un-mark zfs storage driver as FreeBSD-only.

diff --git a/configure.ac b/configure.ac
index 3c5cbf9..7d74f53 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2011,14 +2011,6 @@ if test "$with_storage_gluster" = "yes"; then
 fi
 AM_CONDITIONAL([WITH_STORAGE_GLUSTER], [test "$with_storage_gluster" = "yes"])
 
-if test "$with_storage_zfs" = "check"; then
-    with_storage_zfs=$with_freebsd
-fi
-
-if test "$with_storage_zfs" = "yes" && test "$with_freebsd" = "no"; then
-    AC_MSG_ERROR([The ZFS storage driver can be enabled on FreeBSD only.])
-fi
-
 if test "$with_storage_zfs" = "yes" ||
    test "$with_storage_zfs" = "check"; then
   AC_PATH_PROG([ZFS], [zfs], [], [$PATH:/sbin:/usr/sbin])
